### PR TITLE
Fix for #504 - surface correct validation error messages if save…

### PR DESCRIPTION
**Pull request by theirishpenguin**
... fails in an after_save block.

Note: Need to add tests, if this is deemed to be in the right direction.

### CODE REVIEW

**Comment by lgebhardt** [comment]
I'm ok with this change even though it's a non-normal situation. Could you please rebase against master and add a test for the new code (check for 100% coverage using `COVERAGE=true rake test` please).

**Comment by theirishpenguin** [comment]
Great @lgebhardt. I hope to get some time this week to complete this and will do the actions listed in your comment. Thanks for the quick reply.

**Comment by theirishpenguin** [comment]
Hi @lgebhardt, I have rebased and updated this PR with a test.

I'm not familiar with 100% of the jsonapi-resource codebase so I may not have implemented the test in the correct style - is it okay?

File: test/unit/resource/resource_test.rb [hunk 1]
```diff
@@ -8,6 +8,25 @@ def self.records(options)
...
+  end
+end
+
+class ArticleWithBadAfterSaveResource < JSONAPI::Resource
+  model_name 'PostWithBadAfterSave'
+  attribute :title
+
+  def self.records(options)
+    options[:context].posts
+  end
```
Comment (lgebhardt) [inline test/unit/resource/resource_test.rb]: This method isn't being called in the tests. Could you please remove it?

**Comment by lgebhardt** [comment]
Other than the one uncovered line things look good. Thanks.

**Comment by theirishpenguin** [comment]
Cool. Done, thanks.

**Comment by lgebhardt** [comment]
Thanks!

**Comment by theirishpenguin** [comment]
Great, thanks!

**Pull request merged**